src/mito2/src/region/opener.rs

```diff
@@ -249,12 +249,28 @@ impl RegionOpener {
     }
 
     /// Returns a new manifest options.
-    fn manifest_options(&self, config: &MitoConfig) -> RegionManifestOptions {
-        RegionManifestOptions {
+    fn manifest_options(&self, config: &MitoConfig) -> Result<RegionManifestOptions> {
```

Comment:
Let's change the `manifest_options` signature to accept `&RegionOptions`. Then we don't need to create another `RegionOptions` in this function. It looks like all the places calling this function have a `RegionOptions`.